posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -337,9 +337,40 @@ async def materialize_model(model_label: str, team: Team) -> tuple[str, DeltaTab
         prepare_s3_files_for_querying(saved_query.folder_path, saved_query.name, file_uris)
 
     key, delta_table = tables.popitem()
+
+    row_count = await asyncio.to_thread(count_delta_table_rows, delta_table)
+    await update_table_row_count(saved_query, row_count)
+
     return (key, delta_table)
 
 
+def count_delta_table_rows(delta_table: DeltaTable) -> int:
+    """
+    Count the number of rows in a Delta table using metadata.
+    """
+    count = 0
+    for batch in delta_table.to_pyarrow_dataset().to_batches():
+        count += len(batch)
```

Comment:
OOHH lets not do this - it'll load the whole delta table into memory, likely causing pods to OOM, depending on the size of the saved query, but considering this will likely be used by users with a lot of data to speed up queries, I imagine the tables wont be slow. 

Instead, query the data via clickhouse `select count(*) from ...` - this is what we do in the data import pipeline. A `DataWarehouseTable` model has a `get_count()` method for exactly this